### ISSUE TITLE
Scanning barcode is broken

### ISSUE THREAD

**Issue by TheBird956** [grocy/grocy#857]
Whenever I am clicking on the photo icon in the barcode field to scan a barcode, the popup shows up but nothing work and I get this error in the console:

```log
Error: enumerateDevices is not defined
    at r (quagga.min.js?v=2.7.1:1)
    at Object.u [as enumerateVideoDevices] (quagga.min.js?v=2.7.1:1)
    at HTMLAnchorElement.<anonymous> (barcodescanner.js?v=2.7.1:253)
    at HTMLDocument.dispatch (jquery.min.js?v=2.7.1:2)
    at HTMLDocument.v.handle (jquery.min.js?v=2.7.1:2)
```

When I try to close the window with cancel or X I get this error:

```log
Uncaught TypeError: Cannot read property 'type' of undefined
    at Object.stop (quagga.min.js?v=2.7.1:1)
    at Object.Grocy.Components.BarcodeScanner.StopScanning (barcodescanner.js?v=2.7.1:135)
    at k.fn.init.onEscape (barcodescanner.js?v=2.7.1:223)
    at k (bootbox.min.js?v=2.7.1:6)
    at HTMLButtonElement.<anonymous> (bootbox.min.js?v=2.7.1:6)
    at HTMLDivElement.dispatch (jquery.min.js?v=2.7.1:2)
    at HTMLDivElement.v.handle (jquery.min.js?v=2.7.1:2)
```

The lightbulb button (that I don't know what is supposed to be since I just started using Grocy) does not do anything, no error.

I am running version 2.7.1, installed on Unraid. I tried with Brave v1.8.96, Firefox v76.0.1 and Brave Mobile v1.8.112

**Comment by puneit-hassio** [comment]
You can try using the app Pantry party. That works for me

**Comment by TheBird956** [comment]
> You can try using the app Pantry party. That works for me 

That's what I've been using but the features are very limited and I'm exploring the idea to have a dedicated device/scanner/screen in the future that will not be an android phone.

**Comment by berrnd** [comment]
Camera barcode scanning depends on [QuaggaJS](https://github.com/serratus/quaggaJS) which uses browser APIs to make that happen. From the error message it seems your browser doesn't support that, or maybe grocy is not served via a secure (`https://`) connection, most browsers only allow camera access on secure connection.

It definitely works an Firefox and Chrome, but I don't know much (and test) about other browsers...

**Comment by TheBird956** [comment]
That was the problem, I was connected through HTTP. Once I got in with HTTPS it started working on my laptop. For some reason I was not getting any message telling me I should connect through HTTPS.

However, on Brave Mobile I get the permission popup for the camera, but when I tap on Allow or Block, nothing happens. I had to manually allow the camera to be used in the browser settings. It worked without problem on Firefox Mobile,
